packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -354,6 +356,20 @@ class XcodeProjectInterpreter {
   }
 }
 
+/// Environment variables prefixed by FLUTTER_XCODE_ will be passed as build configurations to xcodebuild.
+/// This allows developers to pass arbitrary build settings in without the tool needing to make a flag
+/// for or be aware of each one. This could be used to set code signing build settings in a CI
+/// environment without requiring settings changes in the Xcode project.
+List<String> environmentVariablesAsXcodeBuildSettings() {
+  const String xcodeBuildSettingPrefix = 'FLUTTER_XCODE_';
+  return platform.environment.entries
+    .where(
+      (MapEntry<String, String> mapEntry) => mapEntry.key.startsWith(xcodeBuildSettingPrefix))
+    .expand<String>(
+      (MapEntry<String, String> mapEntry) => <String>['${mapEntry.key.substring(xcodeBuildSettingPrefix.length)}=${mapEntry.value}'])
```

Comment:
Could you refactor this a little? I know you have tests for this, but it's difficult to read now, and I worry that later on contributors might make one or two little tweaks to it...